File: app/src/main/java/com/example/tap_pass/RequestAdapter.java
package com.example.tap_pass;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

public class RequestAdapter extends RecyclerView.Adapter<RequestAdapter.RequestViewHolder> {

    private List<Request> requestList;
    private Context context;

    public RequestAdapter(Context context, List<Request> requestList) {
        this.context = context;
        this.requestList = requestList;
    }

    @NonNull
    @Override
    public RequestViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        // Corrected to use the admin request card layout
        View view = LayoutInflater.from(context).inflate(R.layout.item_request_card, parent, false);
        return new RequestViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RequestViewHolder holder, int position) {
        Request request = requestList.get(position);

        // Populate the views based on item_request_card.xml
        // Note: You will need to ensure your Request model class has getters for this data.
        holder.txtName.setText("Name: " + request.getUserName());
        holder.txtAmount.setText("Amount: ₱500.00"); // Placeholder - you can get this from the request object
        holder.txtRfid.setText("RFID: 123456789");   // Placeholder - you can get this from the request object

        // Use a placeholder for the receipt image
        holder.imgReceipt.setImageResource(android.R.drawable.ic_menu_gallery); // A standard placeholder

        // Set click listeners for the Send (approve) and Reject buttons
        holder.btnSend.setOnClickListener(v -> {
            String message = "Request for " + request.getUserName() + " approved.";
            Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
            // Logic to handle approval would go here
        });

        holder.btnReject.setOnClickListener(v -> {
            String message = "Request for " + request.getUserName() + " rejected.";
            Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
            // Logic to handle rejection would go here
        });
    }

    @Override
    public int getItemCount() {
        return requestList.size();
    }

    // ViewHolder updated to match the views in item_request_card.xml
    public static class RequestViewHolder extends RecyclerView.ViewHolder {
        ImageView imgReceipt;
        TextView txtName;
        TextView txtAmount;
        TextView txtRfid;
        Button btnSend;
        Button btnReject;

        public RequestViewHolder(@NonNull View itemView) {
            super(itemView);
            imgReceipt = itemView.findViewById(R.id.imgReceipt);
            txtName = itemView.findViewById(R.id.txtName);
            txtAmount = itemView.findViewById(R.id.txtAmount);
            txtRfid = itemView.findViewById(R.id.txtRfid);
            btnSend = itemView.findViewById(R.id.btnSend);
            btnReject = itemView.findViewById(R.id.btnReject);
        }
    }
}

File: app/src/main/java/com/example/tap_pass/admin/RequestAdapter.kt
package com.example.tap_pass.admin

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.tap_pass.R

class RequestAdapter(
    private val context: Context,
    private val requestList: List<TopUpRequest>
) : RecyclerView.Adapter<RequestAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val imgReceipt: ImageView = view.findViewById(R.id.imgReceipt)
        val txtName: TextView = view.findViewById(R.id.txtName)
        val txtAmount: TextView = view.findViewById(R.id.txtAmount)
        val txtRfid: TextView = view.findViewById(R.id.txtRfid)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_request_card, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val request = requestList[position]

        holder.txtName.text = "Name: ${request.fullName}"
        holder.txtAmount.text = "Amount: ₱${request.amount}"
        holder.txtRfid.text = "RFID: ${request.rfidUid}"

        // Load receipt preview using Glide
        if (request.proof.isNotEmpty()) {
            Glide.with(context)
                .load(request.proof)
                .centerCrop()
                .placeholder(android.R.drawable.ic_menu_gallery)
                .into(holder.imgReceipt)
        }

        // REDIRECT LOGIC
        holder.itemView.setOnClickListener {
            val detailFragment = RequestDetailFragment()
            val bundle = Bundle()
            bundle.putSerializable("selected_request", request)
            detailFragment.arguments = bundle

            if (context is AppCompatActivity) {
                // Using R.id.nav_host_fragment to match your RelativeLayout XML
                context.supportFragmentManager.beginTransaction()
                    .setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out, android.R.anim.fade_in, android.R.anim.fade_out)
                    .replace(R.id.nav_host_fragment, detailFragment)
                    .addToBackStack(null)
                    .commit()
            }
        }
    }

    override fun getItemCount() = requestList.size
}